tests/tests_core/test_outliers.py

```diff
@@ -1,14 +1,75 @@
 from __future__ import division
-from expan.core.experiment import Experiment
+from expan.core.experiment import Experiment, _choose_threshold_type
 import pandas as pd
 import numpy as np
 
-def test_quantile_filtering():
+
+def test_choose_threshold_type_upper():
+    data = np.array([1, 2, 3])
+    assert _choose_threshold_type(data) == 'upper'
+
+
+def test_choose_threshold_type_upper_zero():
+    data = np.array([0, 1, 2, 3])
+    assert _choose_threshold_type(data) == 'upper'
+
+
+def test_choose_threshold_type_lower():
+    data = np.array([-3, -2, -1])
+    assert _choose_threshold_type(data) == 'lower'
+
+
+def test_choose_threshold_type_lower_zero():
+    data = np.array([-3, -2, -1, 0])
+    assert _choose_threshold_type(data) == 'lower'
+
+
+def test_choose_threshold_type_two_sided():
+    data = np.array([-3, -2, -1, 0, 1, 2, 3])
+    assert _choose_threshold_type(data) == 'two-sided'
+
+
```

Comment:
Let's add a test with a `nan`, e.g. `data = np.array([-3, np.nan, -1])`

(This comment is related to my comment on the implementation of `_choose_threshold_type`)

I made such a test on my laptop and I see the message: `ValueError: Both min and max of data are zero.`. That message isn't true of course, but it seems to be how `np.nan` are interpreted in the current implementation

Reply:
I don't throw this error anymore

Reply:
There should be at least one test with a `nan` in it. Whatever `NaN`-behaviour we have should be tested (and documented?)

Currently, it would simply be something like: 

```
def test_choose_threshold_type_with_nan():
    # confirm that any data with any 'np.nan' in it will result in 'two-sided'
    data = np.array([-3, -2, -1, np.nan])
    assert _choose_threshold_type(data) == 'two-sided'
```

Reply:
Added `nan`s to tests.

Reply:
Nice. Resolving